training-material/checklist.md

```diff
@@ -0,0 +1,104 @@
+# A proof checklist for writers and reviewers
+
+This is a check list of best practices intended
+
+* for proof writers to use before checking a proof into a repository and
+* for proof reviewers to use during the code review of a pull
+  request containing a proof.
+
+The best practices for writing a proof are already described in detail
+in the section [How to Write a CBMC Proof](PROOF-WRITING.md)
+of the [training material](README.md)
+win this [starter kit](../README.md).
+
+There are two questions whose answers must be absolutely clear:
+
+* What properties are being checked by the proof?
+* What assumptions are being made by the proof?
+
+These answers should be found in one of three places:
+
+* The proof harness,
+* The proof makefiles (and, with the starter kit, these makefiles are
+  the proof Makefile, the project Makefile-project-defines,
+  and the project Makefile.common), and perhaps
+* The proof readme file.
+
+## Properties checked
+
```

Comment:
It is a stylistic thing, but the current doc reads as best practices, not a checklist.  For example: 

* Document the decision to omit one of these property-checking flags.
  There are valid reasons to omit them either for a project or for an
  individual proof. But the decision and the reason for the decision
  must be documented either in a project readme or a proof readme file.

How about instead we have: 
* All deviations from the standard property-checking flags are documented.

  There are valid reasons to omit them either for a project or for an
  individual proof. But the decision and the reason for the decision
  must be documented either in a project readme or a proof readme file.

The latter form is something that can easily be "checked off".